src/core/execute.h

```diff
@@ -247,6 +247,7 @@ struct ExecParameters {
         int *fds;
         char **fd_names;
         unsigned n_fds;
+        unsigned n_sock_fds;
```

Comment:
i really don't like the abbreviation "n_sock_fds" instead of "n_socket_fds" I must say (really, abbreviating 2 characters away?). Also, I don't like the redundancy of this, i.e. that n_fds includes n_sock_fds.

I think these should be two independent counters:

unsigned n_socket_fds;
unsigned n_storage_fds;

And that this shoul be passed everywhere like this. And if code wants to iterate through all of them it should explicitly add n_socket_fds + n_storage_fds. That way the redundancy goes away as each fd is counted at one place only. (The array for storing the fds can stay together)

hope this makes sense?